optuna/study/study.py

```diff
@@ -45,6 +50,51 @@
 _logger = logging.get_logger(__name__)
 
 
+_T = TypeVar("_T")
+_P = ParamSpec("_P")
+
+
+def _convert_positional_args(
+    *,
+    previous_positional_arg_names: Sequence[str],
+) -> Callable[[Callable[_P, _T]], Callable[_P, _T]]:
+    def decorator(func: Callable[_P, _T]) -> Callable[_P, _T]:
+        assert (
+            previous_positional_arg_names
+            == list(signature(func).parameters)[: len(previous_positional_arg_names)]
+        ), (
+            f"{previous_positional_arg_names} is not a sublist of"
+            f" {list(signature(func).parameters)}"
+        )
```

Comment:
How about allowing `func` to change the order of the (keyword only) arguments? That is, relax the assertion to only check that `previous_positional_arg_names` is a subset.

Reply:
Sure. Relax the check in f5103d5.